KPI: add new placeholder build to handle processing of Allure data.

worldomonation opened the pull request:
#### Proposed Changes

This PR adds a new placeholder build that is intended to process the generated Allure results.
This PR also fleshes out the KPIDashboardTests experimental build configuration to reflect how a future Pre-Release Tests build may look like.

Key changes:
- addition of params, steps and artifact rules for the KPIDashboardTests.
- addition of new project "Calypso Dashboard" with a new build configuration `CalypsoPreReleaseDashboard` which will be tasked with processing and uploading of the report.
- new secrets to interact with AWS S3 bucket.

#### Testing Instructions
Ensure the following build configurations are passing:
  - [x] Code Style
  - [x] Unit Tests

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

worldomonation committed:
Base commit

worldomonation committed:
After discussion with Dan:
- add new build chain after KPIDashboardTests as experiment
- rely on build finish trigger

worldomonation commented:
FYI, in the process of making this PR the TeamCity configuration got messed up: context in pMz3w-gcN-p2.

dpasque commented on .teamcity/settings.kts:
```diff
@@ -82,6 +83,12 @@ project {
+		// Calypso dashboard AWS secrets for S3 bucket.
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_ACCESS_KEY_ID", "credentialsJSON:1f324549-3795-43e5-a8c2-fb81d6e7c15d", display = ParameterDisplay.HIDDEN)
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_SECRET_ACCESS_KEY", "credentialsJSON:782b4bde-b73d-4326-9970-5a79251bdf07", display = ParameterDisplay.HIDDEN)
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_ROOT", "s3://a8c-calypso-e2e-reports")
```
**Question:** Is this a sensitive value? I figure if it's not coming off of the credentials JSON, we probably don't need to use `password`, right? And could just use `param` or `text`?

worldomonation replied:
Right, I had intended to keep it a secret value initially, but it seemed a little unnecessary. Let's switch it back to a text.

dpasque commented on .teamcity/_self/projects/CalypsoDashboard.kt:
```diff
@@ -0,0 +1,51 @@
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildSteps
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.finishBuildTrigger
+
+object CalypsoDashboard : Project({
```
**Suggestion/Question:** Okay, kind of playing devil's advocate here... But should we move this back into the `WebApp` project for now?

I really appreciate you trying to break up the massive behemoth that the `WebApp` project has become... We could certainly use some more granular modularization.

And at the same time, this distinction feels a little weird to me. This is one of the only Calypso related things that isn't in the `WebApp` project, and it is tightly coupled to a build in the `WebApp` project.

What if we just put this in WebApp for now, and do a separate dedicated PR to try to break `WebApp` up into multiple steps?

worldomonation replied:
Done - merged the new build configuration back into WebApp.kt.

dpasque commented on .teamcity/_self/projects/WebApp.kt:
```diff
@@ -874,18 +874,124 @@ object HelpCentreE2ETests : E2EBuildType(
+	features {
+		perfmon {
+		}
+		commitStatusPublisher {
```
**Suggestion:** We can remove this right? We don't need to send this info back to GitHub in any way, do we?

worldomonation replied:
Done - removed the commit publisher.

dpasque reviewed (approved):
Approving! I left a few questions and suggestions, but they aren't blocking and I don't think it will need re-review.

From what I could tell, I wasn't able to test the new sub-project in TeamCity prior to merge, but I did run the updated build.

worldomonation committed:
.teamcity/_self/projects/WebApp.kt
- move CalypsoPreReleaseDashboard build config into WebApp.kt

.teamcity/settings.kts
- change "CALYPSO_E2E_DASHBOARD_AWS_S3_ROOT" type to text

.teamcity/_self/projects/CalypsoDashboard.kt
- remvoe
